File: VirtualGardens/VirtualGardens.Services/Database/Korisnici.cs
﻿using System;
using System.Collections.Generic;

namespace VirtualGardens.Services.Database;

public partial class Korisnici
{
    public int KorisnikId { get; set; }

    public string KorisnickoIme { get; set; } = null!;

    public string Lozinka { get; set; } = null!;

    public string Email { get; set; } = null!;

    public string Ime { get; set; } = null!;

    public string Prezime { get; set; } = null!;

    public string? BrojTelefona { get; set; }

    public string? Adresa { get; set; }

    public string? Grad { get; set; }

    public string? Drzava { get; set; }

    public DateTime DatumRegistracije { get; set; }

    public DateTime? ZadnjiLogin { get; set; }

    public bool JeAktivan { get; set; }

    public DateTime? DatumRodjenja { get; set; }

    public virtual ICollection<KorisniciUloge> KorisniciUloges { get; set; } = new List<KorisniciUloge>();

    public virtual ICollection<Narudzbe> Narudzbes { get; set; } = new List<Narudzbe>();

    public virtual ICollection<PitanjaOdgovori> PitanjaOdgovoris { get; set; } = new List<PitanjaOdgovori>();

    public virtual ICollection<Recenzije> Recenzijes { get; set; } = new List<Recenzije>();

    public virtual ICollection<Ulazi> Ulazis { get; set; } = new List<Ulazi>();
}

File: UI/virtualgardens_admin/lib/models/korisnici.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:virtualgardens_admin/models/korisnici_uloge.dart';

part 'korisnici.g.dart';

@JsonSerializable()
class Korisnik {
  int korisnikId;
  String korisnickoIme;
  String email;
  String ime;
  String prezime;
  String? brojTelefona;
  String? adresa;
  String? grad;
  String? drzava;
  DateTime datumRegistracije;
  DateTime? zadnjiLogin;
  bool jeAktivan;
  DateTime? datumRodjenja;
  String? slika;
  List<KorisnikUloga> korisniciUloges;

  Korisnik(
      {required this.korisnikId,
      required this.korisnickoIme,
      required this.email,
      required this.ime,
      required this.prezime,
      required this.datumRegistracije,
      this.brojTelefona,
      this.adresa,
      this.grad,
      this.drzava,
      this.zadnjiLogin,
      this.jeAktivan = true,
      this.datumRodjenja,
      this.slika,
      this.korisniciUloges = const []});

  factory Korisnik.fromJson(Map<String, dynamic> json) =>
      _$KorisnikFromJson(json);

  Map<String, dynamic> toJson() => _$KorisnikToJson(this);
}
